src/main/java/ch/njol/skript/classes/data/DefaultComparators.java

```diff
@@ -294,6 +298,8 @@ public Relation compare(EntityData e, ItemType i) {
 //				return Relation.get(i.isOfType(Material.SKULL_ITEM.getId(), (short) 1));
 			if (e instanceof BoatData)
 				return Relation.get(((BoatData)e).isOfItemType(i));
+			if (SUPPORTS_CHEST_BOAT && e instanceof ChestBoat)
```

Comment:
Since `e` is entiyData, shouldn't you just check for BoatChestData here?

Reply:
my bad... fixed/pushed